util.go

```diff
@@ -230,8 +230,8 @@ func btsEqualFold(s, p []byte) bool {
 
 	for i := 0; i < n; i++ {
 		x := m + (i << 3)
-		av := uint64(*(*byte)(unsafe.Pointer(&s[x])))
-		bv := uint64(*(*byte)(unsafe.Pointer(&p[x])))
+		av := uint64(s[x])
```

Comment:
I believe we have misunderstanding in the previous discussion :)
So I will explain in details.
Previous version does this:
- treat **first 8 bytes** as uint64([ b1 b2 ... b8 ])
- apply mask to it
- take next **8 bytes**

Current version in PR does this:
- treat **first byte** as uint64([ b1 ])
- apply mask to it
- take next **8 bytes**

So we loose 7 bytes here.

Reply:
What I suggest is just to remove use of this `btsEqualFold()` and `strEqualFold()` and substitute them with `bytes.EqualFold()` and `strings.EqualFold()`.